### PR TITLE
[videoplayer][demuxer] Don't assume/force consecutive numbering of de…

### PR DESCRIPTION
**Pull request by ace20022**
…muxer's streams.

### CODE REVIEW

**Comment by ace20022** [comment]
preliminary work for #9099 @FernetMenta

**Comment by FernetMenta** [comment]
+1
are you ok with https://github.com/xbmc/xbmc/pull/9238 ?

**Comment by ace20022** [comment]
Ofc I forgot the pvt part... I'll look at the other commit tomorrow

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCC.cpp [hunk 1]
```diff
@@ -110,7 +110,21 @@ CDVDDemuxCC::~CDVDDemuxCC()
 
 CDemuxStream* CDVDDemuxCC::GetStream(int iStreamId)
```
Comment (Jalle19) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCC.cpp]: Method could be made const.

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h [hunk 2]
```diff
@@ -311,10 +311,12 @@ class CDVDDemux
   virtual int GetStreamLength() = 0;
 
   /*
-   * returns the stream or NULL on error, starting from 0
+   * returns the stream or NULL on error
    */
   virtual CDemuxStream* GetStream(int iStreamId) = 0;
 
+  virtual const std::vector<CDemuxStream*> GetStreams() const = 0;
```
Comment (tamland) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h]: const (the first) is superfluous in this context

**Comment by ace20022** [comment]
jenkins build this please

**Comment by Jalle19** [comment]
If this gets merged then I take it the helper for PVR addons can be dropped completely? It's there only to enforce the correct order of streams.

**Comment by ace20022** [comment]
I mixed the code for the commits ;(

> If this gets merged then I take it the helper for PVR addons can be dropped completely? It's there only to enforce the correct order of streams.

Hopefully, can you point me to the exact code?

**Comment by ace20022** [comment]
jenkins build this please

**Comment by Jalle19** [comment]
I'll try to find ir for you once I get to a computer.

**Comment by ace20022** [comment]
@Jalle19 @FernetMenta could you please have a look? Especially pvr https://github.com/ace20022/xbmc/commit/4cc799c19ca9385ae72b85e8d56434d8e6bec141 drives me crazy. It works now, but I not sure if it's correct.

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
@FernetMenta good to go in? I tried to use the physical id in the demux client over there: https://github.com/ace20022/xbmc/commit/d04f97c2fe212fb6e01eaa9b01fdca5abfde0a1b
but it does not work since  ```m_IDemux->ReadDemux();``` yields packets with ids starting from 0...

**Comment by FernetMenta** [comment]
@ace20022 fine, sorry for the merge conflict

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
@FernetMenta updated, can I test the inputstreamaddon changes somehow?

jenkins build this please

**Comment by FernetMenta** [comment]
@ace20022 @mapfau will submit inputstream.mpd to the binary addons repo soon. The readme should have instructions and pointers to demo streams.

**Comment by ace20022** [comment]
Okay, I hope that I can finish the pr(s) for the multi stream demuxer next week

**Pull request merged**